### PR TITLE
Ensure that `EvaluatorPreprocessor.opMap` has a null-prototype (issue 17554)

### LINKED ISSUE #17554: Javascript object default properties like `toString` are considered "valid" PDF operators
Test case: [hello.pdf](https://github.com/mozilla/pdf.js/files/14002259/hello.pdf)

Configuration:
- Web browser and its version: **Firefox 122.0b5 (Ubuntu snap)**
- Operating system and its version: **Ubuntu 23.04**
- PDF.js version: **PDF.js: 4.0.347 [76e3e5202]**
- Is a browser extension: **no**

Steps to reproduce the problem:

1. Construct a PDF with a stream containing the token `toString` (see example linked above, line 35)

What is the expected behavior?

- Being an invalid operator, it should be ignored.

What went wrong?

screenshot

Potential operator tokens are used as an index into a map (i.e. a standard Javascript object), where certain properties like `id` and `numArgs` are stored. The presence of an element in this map is checked as follows:

https://github.com/mozilla/pdf.js/blob/fce822cde0fb6bab1d976ffcad5623482e18d446/src/core/parser.js#L1289-L1292

(maybe also in other places, I am not super familiar with the codebase)

The problem is that `possibleCommands` (which seems to equal the result of [`opMap`](https://github.com/mozilla/pdf.js/blob/fce822cde0fb6bab1d976ffcad5623482e18d446/src/core/evaluator.js#L4778)) is a regular Javascript object with the `Object` prototype. This means that it has properties like `toString` and `__proto__`. Hence, in these cases `knownCommands[possibleCommand]` will not equal `undefined`, and the operator is assumed to be valid. This then results in an exception later on when further logic attempts to access this operator's properties.

Variants of such a bug could lead to security problems, but this specific one is pretty harmless. Nevertheless it is good to make use of null-prototype objects for cases like this, and/or use a different way of checking operator validity.

### PR DESCRIPTION
**Pull request by Snuffleupagus**
This accidentally regressed in PR #16956, sorry about that!

### CODE REVIEW

File: src/core/evaluator.js [hunk 1]
```diff
@@ -4780,124 +4780,128 @@ class EvaluatorPreprocessor {
...
-      fals: null,
-      false: null,
-      nu: null,
-      nul: null,
-      null: null,
-    });
+    return shadow(
+      this,
+      "opMap",
+      Object.assign(Object.create(null), {
```
Comment (calixteman) [inline src/core/evaluator.js:4786]: Is it equivalent to `Object.create(null, {...}`) ? If yes it'd be simpler.

**Review by calixteman — APPROVED** [review]
LGTM. Thank you.

File: src/core/evaluator.js (hunk 1 above)
Comment (Snuffleupagus) [reply src/core/evaluator.js:4786]: That was my first thought as well, but unfortunately it causes errors (in the browser):
```
... Property descriptor must be an object, got null.
```

**Review by timvandermeij — APPROVED** [review]
r=me as well, with passing tests. Thank you!

**Pull request merged**